### ISSUE TITLE
BUG : wrong place / location in InaSAFE real time

### ISSUE THREAD

**Issue by samnawi** [inasafe/inasafe#2438]
# Problem
I found some wrong place in the InaSAFE realtime earthquake.

# Detail
Like in the example
1.The epicentrum in Bantul, but the location that shown in the table is in Semarang.
![selection_245](https://cloud.githubusercontent.com/assets/8871958/10272668/a8b0fcaa-6b4d-11e5-9806-4d5e635fba48.png)

2.The earthquake is occured in near of Biak, but the table shown that the location is in Manokwari.
![selection_246](https://cloud.githubusercontent.com/assets/8871958/10272679/caf2627c-6b4d-11e5-9a50-223d0820f3db.png)

I tried to find something about this error, maybe this mistake is occurred because the order of nearby place that appear in the report.

For the first case, the earthquake is occurred in Bantul, but the location is informed in Semarang, because Semarang is on the top of the order.
![selection_247](https://cloud.githubusercontent.com/assets/8871958/10272802/7704b50a-6b4f-11e5-92a7-fc6d70c199b8.png)

For the 2nd case, the nearby place just shown Manokwari in the report.
![selection_248](https://cloud.githubusercontent.com/assets/8871958/10272836/f5ad8436-6b4f-11e5-95a9-1f65567bfd94.png)

The case in above is the example from many things similar like that.

# Proposed Solution
Could we make the location become right?
Maybe we can make the location, refers to BMKG release

# CC
@Charlotte-Morgan @timlinux @lucernae 

Thank you.

**Comment by lucernae** [comment]
@samnawi After tracing the code, yes it apparently Akbar made specifically that it will shows the most impacted cities like you said.
So, are we going to change this behaviour to show the location BMKG put in grid.xml?
Need confirmation first @timlinux @Charlotte-Morgan

**Comment by Charlotte-Morgan** [comment]
Hi all.
Iwan. Thanks for spotting this and your recommendation. Can you please confirm with BNPB (maybe Ibu Dian or Pak Agus) that your recommendation meets their requirements.
+1 to make it relevant to users.
Regards
Charlotte

Charlotte Morgan
Spatial Data Analyst
DMInnovation - Jakarta
GEOSCIENCE AUSTRALIA

Mobile: +62 811 820 4648

On 7 Oct 2015, at 9:51 pm, Rizky Maulana Nugraha <notifications@github.com<mailto:notifications@github.com>> wrote:


@samnawi<https://github.com/samnawi> After tracing the code, yes it apparently Akbar made specifically that it will shows the most impacted cities like you said.
So, are we going to change this behaviour to show the location BMKG put in grid.xml?
Need confirmation first @timlinux<https://github.com/timlinux> @Charlotte-Morgan<https://github.com/Charlotte-Morgan>

-
Reply to this email directly or view it on GitHub<https://github.com/AIFDR/inasafe/issues/2438#issuecomment-146154191>.

Geoscience Australia Disclaimer: This e-mail (and files transmitted with it) is intended only for the person or entity to which it is addressed. If you are not the intended recipient, then you have received this e-mail by mistake and any use, dissemination, forwarding, printing or copying of this e-mail and its file attachments is prohibited. The security of emails transmitted cannot be guaranteed; by forwarding or replying to this email, you acknowledge and accept these risks.
-------------------------------------------------------------------------------------------------------------------------

**Comment by samnawi** [comment]
@Charlotte-Morgan I've confirm with Ibu Dian, and she agreed about it. She says that the location should refers to BMKG release, because the EQ information is come from BMKG also.

**Comment by timlinux** [comment]
@lucernae will you follow up on this? Thanks

**Comment by lucernae** [comment]
Following this up.
What needs to be changed:
- [ ] get the location from the location in grid.xml
- [ ] if any, calculate distance and direction of the earthquake from that location

**Comment by samnawi** [comment]
As an email that i've sent to @lucernae on 17 Feb'16.
Just put in here, for note in this issue, link to download placename with population data.
https://www.dropbox.com/sh/ibhodjntetcfcg1/AABAtobVTWZ3sxeTOD0rPVLpa?dl=0

And I agree with your proposed to calculate distance from the epicenter location.

Thank you.

**Comment by lucernae** [comment]
@samnawi I raised some questions after reading and trying to understand these files.

- If we use this, do we ditch the old population data? (the raster one)
- there are 2 population data, in district polygons we will use total population in that district, then in district city points, we use estimated population in that city points.
- CMIIW, are district polygons data used to calculate affected population by MMI? If so, @timlinux can you explain what to do when the MMI boundaries only intersect with some part of the district polygons? Do we interpolate the population count on the intersected part?
- CMIIW, the city points data are used for nearby cities calculations. So the list of the cities and impacted population is coming from this data.

**Comment by lucernae** [comment]
@samnawi also some comments from your statement.

> And I agree with your proposed to calculate distance from the epicenter location.

I tried to calculate the distance from location provided in grid.xml and the epicenter location. But in reality, sometimes grid.xml does not contains city names of nearby location, for example, it contains PAPUA BARAT, MALUKU UTARA, or something similar along those lines. If this is the case, then I tried to calculate nearest cities myself and use that instead in the report.
